layers/fully_connected.py

```diff
@@ -228,11 +229,9 @@ def get_var_x(self, input):
     def forward(self, input):
         batch_size = input.size(0)
         mean_xin = self.get_mean_x(input)
-        output = mean_xin
-        if self.training:
-            var_xin = self.get_var_x(input)
-            eps = self.get_eps(self.floatTensor(batch_size, self.out_features))
-            output = output.add(var_xin.sqrt().mul(eps))
+        var_xin = self.get_var_x(input)
+        eps = self.get_eps(self.floatTensor(batch_size, self.out_features))
+        output = mean_xin.add(var_xin.sqrt().mul(eps))
```

Comment:
Any specific reason you enabled always sampling by default? Usually we use the deterministic equivalent network, i.e. the mean of the weight distribution.

Reply:
Really? I thought you always use sampling, even during inference. If you don't want to sample, you could fix a realization of the noise and still do `mu + sigma*noise`. It seems weird to me to spend all the effort to train both `mu` and `sigma` parameters but only use `mu` during inference. 

With that said, I can change it back, since it's true that `E[W] = mu`, so ultimately the effect of sampling and averaging the samples would result in cancelling the noise term anyway.